### ISSUE TITLE
Remove checkbox logic in Experimental sections wrt scoring

### ISSUE THREAD

**Issue by selinad** [ClinGen/clincoded#1471]
@erinriggs  @marinadistafano @jennygoldstein @bryceseifert -- please confirm you want to remove the checkbox logic (where checking the box is necessary to activate the scoring section) for each of the following experimental evidence types according to the following description for implementation. thx!:

Currently, the following experimental evidence types have a checkbox that, unless checked, prevents a curator from scoring.  

**Experimental evidence with checkboxes that must be checked for scoring to occur:**

Biochemical Function A
* "Has gene been implicated in same disease?"

Protein Interactions
* "Has this gene or genes been implicated in same disease?"

Expression A
* "Is the gene normally expressed in above tissues?"

Expression B
* "Is expression altered in patients who have disease?"

Rescue (all 4 options)
* "Does wild-type rescue rescue the above phenotype?"

This was originally implemented o guide the curator as to what meets criteria.  However, curators would like to be able to add evidence that might not reach the criteria and score as 0, mark for review, or even place as Contradicts.  The solution discussed on today's call would be to do the following:

* Remove the logic behind the checkbox such that scoring is always an option (retain required fields, including the checkbox)
* Add warning box that not checking this means criteria not met (we can provide warning) and this should be taken into account in the scoring section (we can provide text)
* Allow curators to Score in any manner they want (no change needed to interface if checkbox logic removed)

**Comment by marinadistefano** [comment]
This sounds good to me and I think it consistent with what we discussed on the call today.

**Comment by selinad** [comment]
Thanks @marinadistefano  If one or two others could take a look and confirm, that would be great.

**Comment by ErinRiggs** [comment]
Yes, I agree.

**Comment by jennygoldstein** [comment]
I also agree.

Thanks,


Jenny


________________________________

**Comment by selinad** [comment]
Thanks, @ErinRiggs and @jennygoldstein !

**Comment by selinad** [comment]
@jimmyzhen I'm adding R13 to this ONLY in case it makes sense while you are working on other Experimental items and per our conversation you were going to evaluate how much time it would take.  Please feel free to move to R14.

**Comment by jimmyzhen** [comment]
Hi @selinad,

> Add warning box that not checking this means criteria not met (we can provide warning) and this should be taken into account in the scoring section (we can provide text)

Would you mind providing the copies of text as you described above?

**Comment by wrightmw** [comment]
@selinad 
How about?:
"Note: an unchecked box indicates this criterion has not been met for this evidence and this should be considered during its evaluation"

**Comment by selinad** [comment]
thx @wrightmw!

I think criteria is good here as they are several things they need to answer.

Another version:
"Note: not checking this box indicates this criteria has not been met for this evidence; this should be taken into account during its evaluation"

**Comment by wrightmw** [comment]
@selinad Looks good to me :-)

**Comment by jimmyzhen** [comment]
@selinad, @wrightmw,

Does the yellow warning box consisting of this paragraph of text appear below all those checkboxes affected in this ticket?

**Comment by selinad** [comment]
@jimmyzhen yes.  It's generic and if it's not checked, they have not met criteria for all the items listed at top of ticket.

**Comment by jimmyzhen** [comment]
@selinad, @wrightmw,

Here is what may look like after the adding the yellow warning box:
![image](https://user-images.githubusercontent.com/6956310/39791603-6ae9fab4-52f1-11e8-8546-1d1d8c34a6f4.png)

Please let me know whether it works. Thanks.

**Comment by selinad** [comment]
@jimmyzhen this is great!

I guess I do have 2 minor changes seeing it in context:
- say "Warning" instead of "Note"
- say "not checking the above box....." -- it's a bit far down from the checkbox.

**Comment by jimmyzhen** [comment]
@selinad, thanks for the feedback.

**Comment by selinad** [comment]
Indeed, @jimmyzhen.  Sorry for the changes!

**Comment by jimmyzhen** [comment]
Hi @selinad, @wrightmw,

You may review the fix in the following instance:
https://jz-experimental-checkbox.demo.clinicalgenome.org

**Comment by jimmyzhen** [comment]
@selinad, @wrightmw,

I've spun up a new instance for you to review the ability to save the curation without selecting the checkboxes:
https://1471-jz-experimental.demo.clinicalgenome.org

**Comment by wrightmw** [comment]
@selinad 
But I think it's kind of right that they should not fill in details in the "How has this other gene(s) been implicated in the above disease?:" box if they have not checked yes to "Has this gene(s) been implicated in the above disease?:"

**Comment by selinad** [comment]
thx @wrightmw  I think it might be best just to open up all the fields....they are collecting evidence but no scoring it.  It's hard to predict the different scenarios...

**Comment by wrightmw** [comment]
@selinad Sure... I can see that opening up all fields is certainly easier to implement.

**Comment by jimmyzhen** [comment]
@selinad, @wrightmw,

I've updated the instance with the changes discussed today. Please review again. Thanks.

**Comment by selinad** [comment]
@jimmyzhen so far, so good!

One thing (sorry again) -- if @wrightmw agrees, I think we can take this whole note away now for each experimental evidence type (they are somewhat different depending on evidence type) and just leave the yellow warning box:

"Note: If the interacting gene(s) have not been associated with the disease, the criteria for counting this experimental evidence has not been met and cannot be submitted. Curate new Experimental Data or return to Record Curation page."
(from Protein Interactions)

**Comment by ErinRiggs** [comment]
Sorry to intrude, but I got this message and wanted to weigh in on the suggested warning message.  Instead of "Note: If the interacting gene(s) have not been associated with the disease, the criteria for counting this experimental evidence has not been met and cannot be submitted. Curate new Experimental Data or return to Record Curation page,"  I suggest changing "cannot be submitted" to "cannot be scored," or something along those lines.  The scenario in which we would do this is where we want to document that we evaluated something, but because it did not meet the criteria, we are not scoring it, even though it is still being "submitted" per se as information on the record.

**Comment by selinad** [comment]
Thx for weighing in @ErinRiggs !  My latest suggestion (above) was to remove that text entirely and just keep the yellow warning box, currently below it, which says:

"Warning: not checking the above box indicates this criteria has not been met for this evidence; this should be taken into account during its evaluation."

Does this work?  If not, we'd welcome any edits.  Thanks again!

**Comment by ErinRiggs** [comment]
That sounds perfect @selinad - thank you!

**Comment by selinad** [comment]
OK, super -- we always appreciate any input, so thank you very much @ErinRiggs !

**Comment by jimmyzhen** [comment]
@selinad, @wrightmw,

I've again updated the instance with the deletion of the described notes as well as fix for error Selina reported. Please review again. Thank you!

**Comment by selinad** [comment]
@jimmyzhen looks great -- notes gone for each experimental evidence type and yellow box exists correctly.  Perfect, thank you.

Also, the bug is gone.  I didn't do an exhaustive check of this on experimental -- was this specific to cell culture for some reason?

**Comment by jimmyzhen** [comment]
@selinad, the bug is specific to cell culture in **Rescue** because I missed to update the property names in the **Rescue** object in the process of loading and parsing the saved curation data. As a result, the application was not aware of the previously saved UBERON/EFO/CL IDs.

**Comment by jimmyzhen** [comment]
@selinad, btw, I've just tried going through other Experimental types and subtypes. I didn't come across the bug described above.

**Comment by selinad** [comment]
@jimmyzhen everything looking great until I got to Rescue, non-human model organism.  I didn't click the checkbox does wild-type rescue..., but did click the other one - then had the spinning wheel:

**Comment by selinad** [comment]
@jimmyzhen let me know if you need the above to be moved into a different ticket.

All EFO and CL entries worked perfectly! (colon and underscore)

**Comment by jimmyzhen** [comment]
As discovered in #1683, the reported issue was not reproducible.
